src/test/java/pl/matidominati/GitHubApp/controller/GitHubControllerTest.java

```diff
@@ -0,0 +1,111 @@
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import pl.matidominati.GitHubApp.client.GitHubClient;
+import pl.matidominati.GitHubApp.client.model.GitHubOwner;
+import pl.matidominati.GitHubApp.client.model.GitHubRepository;
+import pl.matidominati.GitHubApp.model.dto.RepositoryResponseDto;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+public class GitHubControllerTest {
+
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Autowired
+    private MockMvc mockMvc;
+    @MockBean
+    private GitHubClient client;
+
+    @Test
+    void shouldGetGitHubRepositoryDetails() throws Exception {
+        GitHubRepository gitHubRepository = GitHubRepository.builder()
+                .name("aaaaa")
+                .fullName("bbbbb")
+                .description("ccccc CCCCC")
+                .createdAt(LocalDateTime.of(2015, 10, 10, 10, 10))
+                .cloneUrl("bbbb.com")
+                .id(1L)
+                .gitHubOwner(GitHubOwner.builder()
+                        .id(1L)
+                        .login("user")
+                        .build())
+                .build();
+
+        when(client.getRepositoryDetails(any(), any())).thenReturn(Optional.ofNullable(gitHubRepository));
+
+        assert gitHubRepository != null;
+        MvcResult mvcResult = mockMvc.perform(get("/repositories/{owner}/{repoName}", gitHubRepository.getGitHubOwner().getLogin(), gitHubRepository.getName()))
+                .andDo(print())
+                .andExpect(status().is(200))
+                .andReturn();
+
+        RepositoryResponseDto responseDto = objectMapper.readValue(mvcResult.getResponse().getContentAsString(), RepositoryResponseDto.class);
+        assertThat(responseDto).isNotNull();
+        assertThat(responseDto.getFullName()).isEqualTo(gitHubRepository.getFullName());
+        assertThat(responseDto.getDescription()).isEqualTo(gitHubRepository.getDescription());
+        assertThat(responseDto.getCloneUrl()).isEqualTo(gitHubRepository.getCloneUrl());
+        assertThat(responseDto.getCreatedAt()).isEqualTo(gitHubRepository.getCreatedAt());
+        assertThat(responseDto.getStars()).isEqualTo(gitHubRepository.getStargazersCount());
+    }
+
+    @Test
+    void shouldGetClientRepositoriesList() throws Exception {
+
+        GitHubOwner gitHubOwner = new GitHubOwner(1L, "user");
+
+        GitHubRepository gitHubRepositoryA = GitHubRepository.builder()
```

Comment:
Wynies do metody, ktora to tworzy w zaelznosci od tego jakie parametry przekazesz bo 3 raz po 10 linijek piszesz